src/diffusers/models/controlnet_xs.py

```diff
@@ -1335,13 +1350,16 @@ def get_first_cross_attention(block):
         temb_channels = base_downblock.resnets[0].time_emb_proj.in_features
         num_groups = base_downblock.resnets[0].norm1.num_groups
         ctrl_num_groups = ctrl_downblock.resnets[0].norm1.num_groups
+        use_linear_projection = True # default SD2.1 set linear projection to True
         if hasattr(base_downblock, "attentions"):
             has_crossattn = True
             transformer_layers_per_block = len(base_downblock.attentions[0].transformer_blocks)
             base_num_attention_heads = get_first_cross_attention(base_downblock).heads
             ctrl_num_attention_heads = get_first_cross_attention(ctrl_downblock).heads
             cross_attention_dim = get_first_cross_attention(base_downblock).cross_attention_dim
             upcast_attention = get_first_cross_attention(base_downblock).upcast_attention
+            if hasattr(base_downblock.attentions[0], "use_linear_projection"):
```

Comment:
Is there a case where `base_downblock.attentions[0]` doesn't have a `use_linear_projection` attribute?

Reply:
I was considering when `dual_cross_attention` is set to True in `CrossAttnDownBlock2D`, the `DualTransformer2DModel` has no attribute `use_linear_projection `. But I don't think it is a big concern for released SD models. So I updated the code to make it more consistent with other parameters now.

Reply:
Also looks good!